### PR TITLE
dev/core#4826 Attempt at adding is_new to membership status

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#4826 Attempt at adding is_new to membership status

See  https://lab.civicrm.org/dev/core/-/issues/4826

I didn't get this working & am not currently working further on it. I put this up to show where I got to / what I was thinking 

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@colemanw I took it a bit further but wasn't sure about the prefixing
```
SELECT `a`.`id` AS `id`, IF ({id} = 1, 1, 0) AS `status_id.is_new`
FROM civicrm_membership a
LEFT JOIN `civicrm_membership_status` `status_id_1` ON `a`.`status_id` =  `status_id_1`.`id`
WHERE (`a`.`is_test` = "0")
 [nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '} = 1, 1, 0) AS `status_id.is_new`
FROM civicrm_membership a
LEFT JOIN `civic...' at line 1]
```

**Comment by mattwire** [comment]
I've commented on the lab issue. Not sure that this is a good idea as the `name` field should not really be changed.

**Comment by eileenmcnaughton** [comment]
The ability to remove the 'new' status is really long-standing - I don't think we can change that cos I have definitely seen sites that have removed it in favour of an extended 'current' or similar. It's not reserved so it is what it is.

**Comment by mattwire** [comment]
Right @eileenmcnaughton I understand. I'm focussing on the title/frontend_title first but can come back to this.

File: ext/civi_member/Civi/Api4/Service/Spec/Provider/MembershipStatusGetSpecProvider.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  /**
+   * Determine if the membership status is the status used for new memberships.
+   *
+   * @param array $id
+   * @param \Civi\Api4\Query\Api4SelectQuery $query
+   * return string
+   */
+  public static function isNewMembership(array $fieldSpec, Api4SelectQuery $query): string {
+    $newID = \CRM_Member_BAO_MembershipStatus::getNewMembershipTypeID();
+    return "IF ({id} = $newID, 1, 0)";
```
Comment (colemanw) [inline ext/civi_member/Civi/Api4/Service/Spec/Provider/MembershipStatusGetSpecProvider.php:62]:
```suggestion
    if ($newID) {
      return "IF ({$fieldSpec['sql_name']} = $newID, 1, 0)";
    }
    else {
      return '0';
    }
```

**Comment by colemanw** [comment]
Sorry it took me a while to get back to this @eileenmcnaughton - it's been an intense week.
This callback function receives a complete field spec including the `"sql_name"` which is the rendered sql expression including the prefix and backticks.

**Pull request merged**